File: src/test/java/com/github/klingsbo/greeting/GreetingResourceTest.java
package com.github.klingsbo.greeting;

import io.quarkus.test.common.http.TestHTTPEndpoint;
import io.quarkus.test.junit.QuarkusTest;
import org.junit.jupiter.api.Test;
import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;

@QuarkusTest
@TestHTTPEndpoint(GreetingResource.class)
public class GreetingResourceTest {

    @Test
    public void testHelloEndpoint() {
        given().when().get().then().statusCode(200).body(is("Hello from RESTEasy Reactive"));
    }

    @Test
    public void testPost() {
        given().when().post().then().statusCode(201);
    }

    @Test
    public void newTestHelloEndpoint() {
        given().when().put().then().statusCode(200).body("message", equalTo("New Hello"));
    }
}

File: src/test/kotlin/com/github/klingsbo/greeting/GreetingResourceTest.kt
package com.github.klingsbo.greeting

import io.quarkus.test.common.http.TestHTTPEndpoint
import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured
import org.hamcrest.CoreMatchers
import org.junit.jupiter.api.Test

@QuarkusTest
@TestHTTPEndpoint(GreetingResource::class)
class GreetingResourceTest {
    @Test
    fun testHelloEndpoint() {
        RestAssured.given().`when`().get().then().statusCode(200).body(CoreMatchers.`is`("Hello from RESTEasy Reactive"))
    }

    @Test
    fun testPost() {
        RestAssured.given().`when`().post().then().statusCode(201)
    }

    @Test
    fun newTestHelloEndpoint() {
        RestAssured.given().`when`().put().then().statusCode(200).body("message", CoreMatchers.equalTo("New Hello"))
    }
}